tools/testing/e2e/test_utils.h

```diff
@@ -0,0 +1,153 @@
+#ifndef IREE_TOOLS_TESTING_E2E_TEST_UTILS_H
+#define IREE_TOOLS_TESTING_E2E_TEST_UTILS_H
+#include <stdio.h>
+
+#include "iree/base/api.h"
+#include "iree/base/internal/flags.h"
+#include "iree/hal/api.h"
+#include "iree/vm/api.h"
+
+IREE_FLAG(bool, require_exact_results, true,
+          "Requires floating point result elements to match exactly.");
+IREE_FLAG(
+    float, acceptable_fp_delta, 1e-5f,
+    "Maximum absolute difference allowed with inexact floating point results.");
+IREE_FLAG(
+    int32_t, max_elements_to_check, 10000,
+    "Maximum number of tensor elements to check for the given test. For larger "
+    "buffers, only every n-th element will be checked for some n chosed to "
+    "stay just under that threshold and to avoid being a divisor of the inner "
+    "dimension size to avoid special patterns. As the check uses a slow "
+    "reference implementation, this is a trade-off between test latency and "
+    "coverage. The value 0 means check all elements.");
+
+const char* emoji(bool good);
+
+int calculate_check_every(iree_hal_dim_t tot_elements,
+                          iree_hal_dim_t no_div_of);
```

Comment:
These function in this file should all be namespaced with an `iree_` in their names

Reply:
done.